File: scala/src/main/scala/io/devorbit/sparklabs/batch/sales/SparkSalting.scala
package io.devorbit.sparklabs.batch.sales

import io.devorbit.sparklabs.batch.sales.Utils
import org.apache.spark.sql.expressions.Window
import org.apache.spark.sql.functions._

object SparkSalting extends App {

  val df2 =
    Utils.sparkSession.range(20)
      .withColumn("salt", (rand() * 10).cast("int"))
      .show(false)

//  Thread.sleep(1000000)
}

File: python/sparklabs/batch/sales/sql/spark_salting.py
from sparklabs.batch.sales.sql import utils


def run() -> None:
    utils.sql(
        """
        SELECT id, CAST(rand() * 10 AS INT) AS salt
        FROM RANGE(20)
        """
    ).show(truncate=False)


if __name__ == "__main__":
    run()
